aws/validators.go

```diff
@@ -916,6 +916,84 @@ func validateAwsEmrInstanceGroupRole(v interface{}, k string) (ws []string, erro
 	return
 }
 
+func validateAwsEmrSpotProvisioningTimeOutAction(v interface{}, k string) (ws []string, errors []error) {
+	validTypes := map[string]struct{}{
+		"SWITCH_TO_ON_DEMAND": {},
```

Comment:
You could use the constant `emr.SpotProvisioningTimeoutActionSwitchToOnDemand` here and there are a couple of more places in this file and other files of this PR where you could use constants.